### ISSUE TITLE
InsufficientMemoryException when calling PrivilegeExtension.AdjustPrivilege

### ISSUE THREAD

**Issue by cocowalla** [dahall/Vanara#101]
After upgrading from 3.1.1 to 3.1.8, when calling `PrivilegeExtension.AdjustPrivilege`, I get a fatal exception:

```
System.InsufficientMemoryException: Insufficient memory to continue the execution of the program.
   at Vanara.Extensions.InteropExtensions.WriteNoChecks(IntPtr ptr, Object value, Int32 offset, SizeT allocatedBytes)
   at Vanara.Extensions.InteropExtensions.Write(IntPtr ptr, Object value, Int32 offset, SizeT allocatedBytes)
   at Vanara.InteropServices.SafeAnysizeStruct`1.ToNative(T value)
   at Vanara.InteropServices.SafeAnysizeStruct`1..ctor(T& value, String sizeFieldName)
   at Vanara.PInvoke.AdvApi32.AdjustTokenPrivileges(HTOKEN TokenHandle, Boolean DisableAllPrivileges, TOKEN_PRIVILEGES& NewState, TOKEN_PRIVILEGES& PreviousState)
   at Vanara.Security.AccessControl.PrivilegeExtension.AdjustPrivileges(SafeHTOKEN hObj, TOKEN_PRIVILEGES& privileges)
   at Vanara.Security.AccessControl.PrivilegeExtension.AdjustPrivilege(SafeHTOKEN hObj, SystemPrivilege priv, PrivilegeAttributes attr)
   at Vanara.Extensions.ProcessExtension.EnablePrivilege(Process process, SystemPrivilege privilege)
```

The code to reproduce is simply:

```c#
var currentProcess = Process.GetCurrentProcess();
currentProcess.EnablePrivilege(SystemPrivilege.Backup);
```

I see this with Vanara 3.1.8. I tried 3.1.3 as well, but have a different problem - I get a hard crash (debugger doesn't even catch it!), with this in the Windows Event Log:

```
Faulting application name: Acme.Widget.exe, version: 1.2.3.0, time stamp: 0xba320bf1
Faulting module name: ntdll.dll, version: 10.0.18362.418, time stamp: 0x99ca0526
Exception code: 0xc0000374
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
```

Not sure if it matters, but I'm using Windows 10 v1919, build 18363.535. I should add, the system has 30GB+ of free memory.

**Comment by dahall** [comment]
Thanks for finding this. I'm not sure how it got through my unit testing. It is fixed in the code and I'll push today in 3.1.9.

**Comment by cocowalla** [comment]
That was quick, thanks!
